e2e_test/src/coordinator_tests.rs

```diff
@@ -0,0 +1,379 @@
+            println!("stderr: {}", String::from_utf8_lossy(&output.stderr));
+        }
+        // todo check balance
+        // clean env
+        kill_process("cnosdb-meta");
+        kill_process("cnosdb");
+    }
+}
+
+#[allow(dead_code)]
+async fn prepare() {
+    for i in 0..5 {
+        let tenant = format!("tenant{}", i);
+        let db = format!("tenant{}db1", i);
+        let req = command::WriteCommand::CreateTenant(
+            "cluster_xxx".to_string(),
+            tenant.clone(),
+            models::schema::TenantOptions::default(),
+        );
+
+        let cli = client::MetaHttpClient::new("127.0.0.1:8901".to_string());
+        // create tenant
+        let _rsp = cli
+            .write::<command::CommonResp<Tenant>>(&req)
+            .await
+            .unwrap();
+        // db req
+        let req = command::WriteCommand::CreateDB(
+            "cluster_xxx".to_string(),
+            tenant.clone(),
+            DatabaseSchema::new(&tenant, &db),
+        );
+        // create db
+        let rsp = cli
+            .write::<command::TenaneMetaDataResp>(&req)
+            .await
+            .unwrap();
+
+        println!("rsp: {:?}", rsp);
+    }
+}
+
+#[allow(dead_code)]
+async fn start_cluster() -> std::process::Output {
+    let output: Output = Command::new("bash")
+        .args(["-c", "cd ../ && ./run_cluster.sh"])
+        .output()
+        .expect("failed to execute process");
+    output
+}
+
+#[allow(dead_code)]
+fn kill_process(process_name: &str) {
+    let system = System::new_all();
+    for (pid, process) in system.processes() {
+        if process.name() == process_name {
+            println!("{}: {}", pid, process.name());
+            let output = Command::new("kill")
+                .args(["-9", &(pid.to_string())])
+                .output()
+                .expect("failed to execute process");
+            println!("status: {}", output.status);
+        }
+    }
+}
+
+#[allow(dead_code)]
+async fn prepare_data() {
+    let mut handles: Vec<thread::JoinHandle<()>> = vec![];
+    for i in 0..5 {
+        let api: String = format!(
+            "http://127.0.0.1:8902/api/v1/write?tenant=tenant{}&&db=tenant{}db1",
+            i, i
+        );
+        let handle: thread::JoinHandle<()> = thread::spawn(move || {
+            println!("thread {} started", i);
+            for j in 0..100 {
+                let output = Command::new("curl")
+                    .args([
+                        "-i",
+                        "-u",
+                        "root:",
+                        "-XPOST",
+                        &api,
+                        "-d",
+                        &format!("tb1,tag1=tag1,tag2=tag2 field1={}", j),
+                    ])
+                    .output()
+                    .expect("failed to execute process");
+                // assert!(output.status.code() == Some(200));
+                println!("status: {}", output.status);
```

Comment:
If the exit status of the call of `curl` is not 0, there may be something error in network, database, system, or others. You should throw the error here.

For example

```rust
if !output.status.success() {
    let err = format!(
        "Command curl existed with an error ({}): {}",
        output.status,
        String::from_utf8_lossy(&output.stdout),
    );
    return Err(err);
}
```

If you want to use the above example, you should unwrap twice this thread:

```rust
for t in threads {
    t.join().unwrap().unwrap();
}
```